lib/compress/evaluate.js

```diff
@@ -82,6 +82,9 @@ function def_eval(node, func) {
     node.DEFMETHOD("_eval", func);
 }
 
+// Used to propagate a nullish short-circuit signal upwards through the chain.
+const nullish = {};
```

Comment:
This is actually genius.

The AST structure of an AST_Chain is a bit weird, but this wrangles it perfectly.

Reply:
```suggestion
const nullish = Symbol("This AST_Chain is nullish");
```